### DISCUSSION TITLE
nixos-with-flakes/nixos-with-flakes-enabled

### DISCUSSION THREAD

**Discussion by giscus** [ryan4yin/nixos-and-flakes-book#100] [Announcements]
# nixos-with-flakes/nixos-with-flakes-enabled

An unofficial and opinionated book for beginners

https://nixos-and-flakes.thiscute.world/nixos-with-flakes/nixos-with-flakes-enabled

**Comment by Yesterday3896** [comment]
Hi, total newb here. 

As soon as you enter the "sudo nixos-rebuild switch --flake /etc/nixos/flake.nix#<hostname>" command it will start to a ~5 gb download. What's it downloading and Why?

**Reply by ryan4yin** [reply]
1. Your flake is using the latest version of `nixos-23.11`, which is newer than your system currently is, so it's expected to download all the new builds of packages.
2. You may have added some custom configurations that require packages that are not currently included in the system.

**Comment by jkotran** [comment]
Thank you for sharing your time and talent. This book is outstanding.

There's an omission in the /etc/nixos/flake.nix example. The module 'hardware-configuration.nix' is not shown.

I am new to flakes. The example flake.nix above did not import it unless I explicitly listed it as a module. (My configuration.nix imports it when using NixOS channels configuration.)

**Reply by jjramsey** [reply]
Shouldn't `hardware-configuration.nix` be imported when `configuration.nix` is imported, since `configuration.nix` already imports `hardware-configuration.nix`?

**Comment by kogge** [comment]
Thanks for the book! As a newb with computers as just a hobby this was a lifesaver now that I threw myself out the deep end with Nix. Much appreciated m8!

Just a heads up as I am also a newb with git & github and this may not be the right forum; when doing '''sudo nixos-rebuild switch --recreate-lock-file --flake .''' I get the output of: '''warning: '--recreate-lock-file' is deprecated and will be removed in a future version; use 'nix flake update' instead.'''

Regards,
Oskar Andersson

**Reply by ryan4yin** [reply]
Sorry for the delay—thanks for the heads-up! just removed that deprecated command.
